app/models/comment.rb

```diff
@@ -194,6 +194,20 @@ def hide(editor:)
     end
   end
 
+  def destroy_and_log_event(event: {})
+    return false unless destroy
+
+    info_request.log_event(
```

Comment:
Let's verify that this results in the search index being purged. I think it should, since it should destroy the previous comment events, which causes them to be marked for purging. Still, worth running through manually to be sure.